x-pack/plugins/elastic_assistant/server/ai_assistant_data_clients/knowledge_base/create_knowledge_base_entry.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { v4 as uuidv4 } from 'uuid';
+import { ElasticsearchClient, Logger } from '@kbn/core/server';
+
+import {
+  KnowledgeBaseEntryCreateProps,
+  KnowledgeBaseEntryResponse,
+} from '@kbn/elastic-assistant-common';
+import { AuthenticatedUser } from '@kbn/security-plugin-types-common';
+import { getKnowledgeBaseEntry } from './get_knowledge_base_entry';
+import { CreateKnowledgeBaseEntrySchema } from './types';
+
+export interface CreateKnowledgeBaseEntryParams {
+  esClient: ElasticsearchClient;
+  logger: Logger;
+  knowledgeBaseIndex: string;
+  spaceId: string;
+  user: AuthenticatedUser;
+  knowledgeBaseEntry: KnowledgeBaseEntryCreateProps;
+}
+
+export const createKnowledgeBaseEntry = async ({
+  esClient,
+  knowledgeBaseIndex,
+  spaceId,
+  user,
+  knowledgeBaseEntry,
+  logger,
+}: CreateKnowledgeBaseEntryParams): Promise<KnowledgeBaseEntryResponse | null> => {
+  const createdAt = new Date().toISOString();
+  const body = transformToCreateSchema(createdAt, spaceId, user, knowledgeBaseEntry);
+  try {
+    const response = await esClient.create({
+      body,
+      id: uuidv4(),
+      index: knowledgeBaseIndex,
+      refresh: 'wait_for',
+    });
+
+    return await getKnowledgeBaseEntry({
+      esClient,
+      knowledgeBaseIndex,
+      id: response._id,
+      logger,
+      user,
+    });
+  } catch (err) {
+    logger.error(
+      `Error creating Knowledge Base Entry: ${err} with kbResource: ${knowledgeBaseEntry.metadata.kbResource}`
+    );
+    throw err;
+  }
+};
+
+export const transformToCreateSchema = (
+  createdAt: string,
+  spaceId: string,
+  user: AuthenticatedUser,
+  { metadata, text }: KnowledgeBaseEntryCreateProps
+): CreateKnowledgeBaseEntrySchema => {
+  return {
+    '@timestamp': createdAt,
+    created_at: createdAt,
+    created_by: user.profile_uid ?? 'unknown',
```

Comment:
Is `unknown` ever a valid `profile_uid`?

Reply:
It is not, just using it as a sentinel value at the moment as the `authenticatedUser` can be null on the kbDataClient (if for some reason it was not instantiated from a request, or an unauthenticated request).

@YulNaumenko has started putting auth checks on routes (like here in [post_actions_execute](https://github.com/elastic/kibana/blob/31a2719122430cad7693a43d68c45ecc51cd3d0e/x-pack/plugins/elastic_assistant/server/routes/post_actions_connector_execute.ts#L80-L85)), so I think either that or individual checks in the kbDataClient methods that require an auth'd user as you mentioned below would be good here as we discussed. Or perhaps we can make `authenticatedUser` mandatory on the client and just return a null client (which it can be now) if the [authc.getCurrentUser()](https://github.com/elastic/kibana/blob/4e4e1e72a2ce913b0fc41059b6d3eecc22963a40/x-pack/plugins/elastic_assistant/server/routes/request_context_factory.ts#L59) call returns null.

Reply:
If the user is mandatory, how we will make "shared"/public knowledge?

Reply:
Yeah, good point. We could use a skeleton 'system authenticatedUser', but I'm thinking it might be better to leave optional and do null checks as needed. I'll see what this looks like when I get closer to fleshing out the RBAC portions of the API. Thanks Yuliia!